packages/block-library/src/post-content/edit.js

```diff
@@ -45,12 +49,28 @@ function EditableContent( { context = {} } ) {
 		{ id: postId }
 	);
 
+	const entityRecord = useSelect(
+		( select ) => {
+			return select( coreStore ).getEntityRecord(
+				'postType',
+				postType,
+				postId
+			);
+		},
+		[ postType, postId ]
+	);
+
+	const hasInnerBlocks = !! entityRecord?.content?.raw;
```

Comment:
Just looking above at the `blocks` value returned by `useEntityBlockEditor` — would it work to check if `blocks.length` is greater than `0`?

Here's where `blocks` is determined: https://github.com/WordPress/gutenberg/blob/33d84b036592a5bf31af05b7710f3b2b14163dc4/packages/core-data/src/entity-provider.js#L160

Reply:
Aaaah yes that should work! Well spotted.

Reply:
Ok so it seems that `blocks` are also empty on first load, so for a draft post the content is getting replaced with the empty paragraph template we're passing `useInnerBlocksProps` 🤔 I think we'll have to stick with the entity record after all.

Reply:
Ah, gotcha! And it sounds like using `.raw` is probably safe because prior to accessing this, there's already a check to see whether or not the current user is allowed to edit the current entity ([here](https://github.com/WordPress/gutenberg/blob/cbcdd256709126017da2b16db5eba5e54beac07f/packages/block-library/src/post-content/edit.js#L81)) 👍

One last question, what would the difference here between using `getEntityRecord` and `getEditedEntityRecord`? Would the latter be preferable for any reason?

Something I'm finding odd in testing, is that if I access the empty page first, then the placeholder text is visible. However, if in Browse mode I first access a page with content, and then go to select the page without content, for some reason the placeholder text isn't visible until I click on the Post Content block. Here's a quick screengrab of going from a non-empty page to an empty page:

![2023-05-17 15 32 41](https://github.com/WordPress/gutenberg/assets/14988353/4d3c0dab-255a-42f9-9dbf-05ab6407daed)

Reply:
> what would the difference here between using `getEntityRecord` and `getEditedEntityRecord`? Would the latter be preferable for any reason?

I don't _think_ so because here we're only interested in the initial state of the post as we open the editor, before any edits have happened. I guess `getEditedEntityRecord` would be relevant if we wanted to do something at some point after the post content had already been edited.

Reply:
> However, if in Browse mode I first access a page with content, and then go to select the page without content, for some reason the placeholder text isn't visible until I click on the Post Content block.

Hmmm, I can't reproduce that locally. My empty page Post Content blocks don't look as tall as that one though - is there anything about the styling that could be interfering? By default I'd expect Post Content to be the height of its contents:

<img width="1049" alt="Screenshot 2023-05-17 at 4 06 21 pm" src="https://github.com/WordPress/gutenberg/assets/8096000/ee04f98f-7b82-4f28-85bf-a1e2d87d052f">

Reply:
Oh, I keep forgetting about my weird height value. I _think_ that should be unrelated, but in my `theme.json` I have minimum height set to `70vh` in my test environment:

```
			"core/post-content": {
				"dimensions": {
					"minHeight": "70vh"
				},
```

Reply:
I've re-tested with the `minHeight` styles removed, and it's still occurring for me. In the below screengrab I'm reloading on the Pages screen, then select a published page that contains a single Cover block. Then I click on the empty page, which is a published page:

https://github.com/WordPress/gutenberg/assets/14988353/e4e2e02b-0a85-4a29-b087-473684dc6644

If I first go to the published empty page, then it displays the placeholder as expected 🤔

Reply:
Hmmm no, still can't reproduce it. Do you have any browser or CPU throttling going on or something? I can see the paragraph placeholder takes half a second to load, but it always appears regardless of where I'm navigating from.

Reply:
Hrm, no throttling that I'm aware of. It's very strange! It feels like for whatever reason the block isn't being triggered to re-render, or something in this context, I can't quite figure it out 🤔

When I log out the following values everything seems correct in the console:

```
	console.log( 'entityRecord', entityRecord );
	console.log( 'hasInnerBlocks', hasInnerBlocks );
```

And code-wise the change in this PR otherwise looks good to me.

Reply:
Oooooh got it: I can repro by setting the front page to blog. It doesn't happen if front page is set to a static page. Now to try and figure out why 😅